ts/parts/Tooltip.ts

```diff
@@ -1188,202 +1189,281 @@ H.Tooltip.prototype = {
         labels: (string|Array<(boolean|string)>),
         points: Array<Highcharts.Point>
     ): void {
-        var tooltip = this,
-            boxes = [] as Array<Highcharts.Dictionary<any>>,
-            chart = this.chart,
-            ren = chart.renderer,
-            rightAligned = true,
-            options = this.options,
-            headerHeight = 0,
-            headerTop: boolean,
-            tooltipLabel = this.getLabel(),
-            distributionBoxTop = chart.plotTop;
+        const tooltip = this;
+        const {
+            chart: {
+                chartWidth,
+                chartHeight,
+                plotHeight,
+                plotLeft,
+                plotTop,
+                plotWidth,
+                pointer,
+                renderer: ren,
+                scrollablePixelsX = 0,
+                scrollablePixelsY = 0,
+                scrollingContainer: {
+                    scrollLeft,
+                    scrollTop
+                } = { scrollLeft: 0, scrollTop: 0 },
+                styledMode,
+                xAxis: [{ opposite }]
+            },
+            options,
+            options: {
+                distance = 16,
+                positioner
+            }
+        } = tooltip;
+
+        const boundaries = {
```

Comment:
From reading this code it looks like in a scrolling chart the boundaries are the _plot area_, whereas it is the whole chart otherwise? Is that because of the masks? In that case, a comment here would help.